js/challengeClass.js

```diff
@@ -0,0 +1,108 @@
+class Challenge {
+  constructor(
+    id,
+    title,
+    description,
+    type,
+    minParticipants,
+    maxParticipants,
+    rating,
+    image,
+    labels,
+  ) {
+    this.id = id;
+    this.title = title;
+    this.description = description;
+    this.type = type;
+    this.minParticipants = minParticipants;
+    this.maxParticipants = maxParticipants;
+    this.rating = rating;
+    this.image = image;
+    this.labels = labels;
+  }
+
+  createChallengeStars() {
+    // Drawing a stars
+    const rate = this.rating;
+    const challengeRating = document.createElement("div");
+    challengeRating.className = "room__ratings"
+
+    for (let i = 1; i <= 5; i++) {
+      const star = new Image();
+
+      if (i <= Math.floor(rate)) {
+        star.src = "assets/svg/star-filled.svg";
+        star.className = "star filled";
+      } else if (i === Math.ceil(rate) && rate % 1 !== 0) {
+        star.src = "assets/svg/star-half.svg";
+        star.className = "star half";
+      } else {
+        star.src = "assets/svg/star.svg";
+        star.className = "star unfilled";
+      }
+
+      challengeRating.appendChild(star);
+    }
+    return challengeRating;
+  }
+
+  createChallengeCard() {
+    const liItem = document.createElement("li");
+    liItem.id = "room__carousel__slide" + this.id;
+    liItem.setAttribute('tabindex', 0);
+    liItem.className = "challenge__slide challengey challenges" + this.id;
+
+    const imageContainer = document.createElement("div");
+    imageContainer.className = "challenge__image-container"
+
+    const img = document.createElement("img");
+    img.className = "challenge__img";
+    img.src = this.image;
+    img.alt = this.description;
+    img.loading = "lazy";
+    img.style.width = "100%";
+    img.style.height = "100%";
+    img.style.objectFit = "cover";
+    imageContainer.appendChild(img);
+
+    const detailsContainer = document.createElement("div");
+    detailsContainer.className = "challenge__details";
+
+
+    const title = document.createElement("h2");
+    title.className = "challenge__details__title_temporary";
+    title.textContent = this.title + " (" + this.type + ")";
```

Comment:
This is a temporary solution to display the type element.

Reply:
> This is a temporary solution to display the type element.

Reply:
For accessibility purposes, it would be ideal to keep the changes for the type as you did here. You just need to remove the ::after class from rooms.scss.

good solution!